### ISSUE TITLE
Access i-th image after `image_dataset_from_directory`

### ISSUE THREAD

**Issue by tqdo** [keras-team/keras#19027]
I wonder whether there is a way to access the i-th image (its numpy array) after loading the images in with `image_dataset_from_directory`. The reason for this is I would like to investigate the image after model prediction. Thanks

**Comment by deepsea920415** [comment]
of course.
this is my code.

import matplotlib.pyplot as plt

i = 0
for images, labels in train_dataset.take(i + 1):
    i_th_image = images[0]
    i_th_label = labels[0]
    plt.imshow(i_th_image, cmap="gray")
    plt.show()
    print(i_th_label)

here, i is the index that you want to investigate.

![image](https://github.com/keras-team/keras/assets/155721694/9bf56a78-aa39-47f4-96c9-9248e66159ab)

**Comment by innat** [comment]
You can extract the data component first and do whatever next, 

```python
images= []
for image, target in tqdm(training_dataset.unbatch()):
    images.append(image)
images= np.array(images)
print(images.shape)
```

However, I think if we have this [feature](https://github.com/keras-team/keras/issues/19022), it would be much better for larger dataset and better design.

**Comment by innat** [comment]
An alternative solutions could be [this](https://stackoverflow.com/a/76674353/9215780). For your task (`model.predict`), to adopt this solutions, you first create a placeholder and override the `predict_step` to assign data to the placeholder. 

```python
pred_images = tf.Variable([])
your_model = Keras.Sequential(...)

class OverridePredict(keras.Model):
   ...
   def predict_step(self, images):
      ....
      pred_images.assign(images)

customModel = OverridePredict(your_model)
customModel.predict(dataloader)
pred_images.numpy().shape # (len(test_df), h, w, 3)
```
